ctk/common/src/main/kotlin/org/codice/compliance/verification/binding/PostBindingVerifier.kt

```diff
@@ -27,38 +33,80 @@ import org.codice.compliance.SAMLProfiles_4_1_4_5_a
 import org.codice.compliance.attributeNode
 import org.codice.compliance.children
 import org.codice.compliance.debugPrettyPrintXml
+import org.codice.compliance.debugWithSupplier
+import org.codice.compliance.prettyPrintXml
 import org.codice.compliance.recursiveChildren
+import org.codice.compliance.utils.TestCommon
 import org.codice.compliance.utils.TestCommon.Companion.ASSERTION
 import org.codice.compliance.utils.TestCommon.Companion.DESTINATION
 import org.codice.compliance.utils.TestCommon.Companion.EXAMPLE_RELAY_STATE
 import org.codice.compliance.utils.TestCommon.Companion.IDP_ERROR_RESPONSE_REMINDER_MESSAGE
 import org.codice.compliance.utils.TestCommon.Companion.MAX_RELAY_STATE_LEN
+import org.codice.compliance.utils.TestCommon.Companion.NAME
+import org.codice.compliance.utils.TestCommon.Companion.VALUE
 import org.codice.compliance.utils.TestCommon.Companion.acsUrl
-import org.codice.compliance.utils.decorators.IdpPostResponseDecorator
+import org.codice.compliance.utils.extractSamlResponseForm
 import org.codice.security.saml.SamlProtocol.Binding.HTTP_POST
 import org.codice.security.sign.Decoder
 import kotlin.test.assertNotNull
 
-class PostBindingVerifier(private val response: IdpPostResponseDecorator) : BindingVerifier() {
+@Suppress("TooManyFunctions")
+class PostBindingVerifier(override val response: Response) : BindingVerifier(response) {
+    companion object {
+        private const val HIDDEN = "hidden"
+        private const val TYPE = "type"
+        private const val ACTION = "action"
+        private const val METHOD = "method"
+        private const val POST = "POST"
+    }
+
+    private val responseForm: Node?
+    private val samlResponseFormControl: Node?
+    private val samlResponse: String?
+    private val relayStateFormControl: Node?
+    private val relayState: String?
+
+    init {
+        responseForm = response.extractSamlResponseForm()
+        samlResponseFormControl =
+                responseForm
+                        ?.children()
+                        ?.list()
+                        ?.filter {
+                            SAML_RESPONSE.equals(it.attributes().get(
+                                    NAME), ignoreCase = true)
+                        }?.firstOrNull()
+        samlResponse = extractValue(samlResponseFormControl)
+        relayStateFormControl =
+                responseForm
+                        ?.children()
+                        ?.list()
+                        ?.filter {
+                            RELAY_STATE.equals(it.attributes().get(TestCommon.NAME),
+                                    ignoreCase = true)
+                        }?.firstOrNull()
+        relayState = extractValue(relayStateFormControl)
+    }
```

Comment:
☝️ This is basically all the parsing that the Decorator object did in various places.